Doc/library/typing.rst

```diff
@@ -2122,7 +2376,7 @@ Abstract Base Classes
 Corresponding to collections in :mod:`collections.abc`
```

Comment:
Something strange is going on with the rendering in this section, and other sections as well (possibly because Sphinx doesn't support PEP 695 yet?)

Here's what I see in the deploy preview:

<details>

![image](https://github.com/python/cpython/assets/66076021/6eae0607-a7cf-462a-a94c-bf127ff16d3e)

</details>

Compared to the status quo:

<details>

![image](https://github.com/python/cpython/assets/66076021/533375df-1304-4d10-a950-5a44a42f778b)

</details>

It feels like it probably isn't a coincidence that the only one getting rendered correctly is `typing.ByteString`, which doesn't use PEP-695 syntax (because it isn't generic!).

Reply:
That's unfortunate, I might have to go back to the old syntax for now.

Reply:
Opened sphinx-doc/sphinx#11438.

Reply:
We would only be able to use PEP 695 syntax in CPython's documentation when ``needs_sphinx`` is increased to be not less than the first version that included support for the novel syntax. Increasing the minimum version is quite conservative, as we wait for redistributors to increase their minimum supported Sphinx version. For indication, the current minimum required is Sphinx 3, released 3 years ago (July 2020).